### ISSUE TITLE
doesn't build on current gentoo-sources 4.8.1

### ISSUE THREAD

**Issue by nielsd** [toaster/gentoo-overlay#3]
```
>>> Emerging (2 of 2) media-video/bcwc_pcie-9999::Tilos-Gentoo-Overlay
 * Determining the location of the kernel source code
 * Found kernel source directory:
 *     /usr/src/linux
 * Found kernel object directory:
 *     /lib/modules/4.8.1-niels/build
 * Found sources for kernel version:
 *     4.8.1-niels
 * Checking for suitable kernel configuration options...                                                                                                                                                                                                                                                              [ ok ]
>>> Unpacking source...
Initialized empty Git repository in /usr/portage/distfiles/git3-src/patjak_bcwc_pcie.git/
 * Fetching https://github.com/patjak/bcwc_pcie ...
git fetch https://github.com/patjak/bcwc_pcie +HEAD:refs/git-r3/HEAD
remote: Counting objects: 873, done.
remote: Total 873 (delta 0), reused 0 (delta 0), pack-reused 873
Receiving objects: 100% (873/873), 289.24 KiB | 249.00 KiB/s, done.
Resolving deltas: 100% (577/577), done.
From https://github.com/patjak/bcwc_pcie
 * [new ref]                    -> refs/git-r3/HEAD
git symbolic-ref refs/git-r3/media-video/bcwc_pcie/0/__main__ refs/git-r3/HEAD
 * Checking out https://github.com/patjak/bcwc_pcie to /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999 ...
git checkout --quiet refs/git-r3/HEAD
GIT NEW branch -->
   repository:               https://github.com/patjak/bcwc_pcie
   at the commit:            48f9854e5d18ec360f36d9242a575ad072dedab9
>>> Source unpacked in /var/tmp/portage/media-video/bcwc_pcie-9999/work
>>> Preparing source in /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999 ...
ln: failed to create symbolic link 'Module.symvers': File exists
 * Preparing facetimehd module
make -j8 HOSTCC=x86_64-pc-linux-gnu-gcc CROSS_COMPILE=x86_64-pc-linux-gnu- 'LDFLAGS=-m elf_x86_64' all
make -C /usr/src/linux M=/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999 modules
make[1]: Entering directory '/usr/src/linux-4.8.1-niels'
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_ddr.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_hw.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_drv.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_ringbuf.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_isp.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_buffer.o
  CC [M]  /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_debugfs.o
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c:298:17: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .queue_setup            = fthd_buffer_queue_setup,
                 ^
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c:298:17: note: (near initialization for ‘vb2_queue_ops.queue_setup’)
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c: In function ‘fthd_v4l2_register’:
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c:714:24: error: implicit declaration of function ‘vb2_dma_sg_init_ctx’ [-Werror=implicit-function-declaration]
  dev_priv->alloc_ctx = vb2_dma_sg_init_ctx(&dev_priv->pdev->dev);
                        ^
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c:714:22: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  dev_priv->alloc_ctx = vb2_dma_sg_init_ctx(&dev_priv->pdev->dev);
                      ^
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c: In function ‘fthd_v4l2_unregister’:
/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.c:749:2: error: implicit declaration of function ‘vb2_dma_sg_cleanup_ctx’ [-Werror=implicit-function-declaration]
  vb2_dma_sg_cleanup_ctx(dev_priv->alloc_ctx);
  ^
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:289: /var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999/fthd_v4l2.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:1471: _module_/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999] Error 2
make[1]: Leaving directory '/usr/src/linux-4.8.1-niels'
make: *** [Makefile:12: all] Error 2
 * ERROR: media-video/bcwc_pcie-9999::Tilos-Gentoo-Overlay failed (compile phase):
 *   emake failed

```
(even tried to build by hand with same results)

**Comment by toaster** [comment]
looks like patjak/bcwc_pcie#92

**Comment by nielsd** [comment]
ahh, many thanks!

Building with (current) "mainline"- as advised there - seems to work properly. Could make sense to adopt that into the ebuild.

many thanks for your work and time so far!

Niels.

**Comment by toaster** [comment]
Hi Niels,

the master branch includes some other commits that are not on mainline.
I guess that mainline is for unstable development.
As soon as I have time I'll switch the ebuild for version 9999 to mainline and introduce a lower number for master branch.


Tilo

**Comment by toaster** [comment]
Hi Niels,

this should be fixed with f5a2002. I don't have 4.8 (currently I am on 4.5 because 4.7 has issues with suspend on my MBP).
Therefore the module does not build on my system with the mainline branch (I introduced the `master_branch` USE flag one commit later to address this).
Could you please give it a try and close the issue if it builds with 4.8?

Thanks

Tilo

**Comment by mikenerone** [comment]
That "one commit later", f2512e4739b0fcea9f802e9f2c4a5fa9bbae55b6, added a call to `use` in the global scope of the ebuild. `use` is only allowed to be used inside an ebuild function. This currently leads to this error:

```
# emerge -p bcwc_pcie

These are the packages that would be merged, in order:

Calculating dependencies | * ERROR: media-video/bcwc_pcie-9999::toaster failed (depend phase):
 *   use() calls are not allowed in global scope
 * 
 * Call stack:
 *               ebuild.sh, line 606:  Called source '/var/lib/overlays/toaster/media-video/bcwc_pcie/bcwc_pcie-9999.ebuild'
 *   bcwc_pcie-9999.ebuild, line  14:  Called use 'master_branch'
 *               ebuild.sh, line  78:  Called die
 * The specific snippet of code:
 *              eval "${x}() { die \"\${FUNCNAME}() calls are not allowed in global scope\"; }"
 * 
 * If you need support, post the output of `emerge --info '=media-video/bcwc_pcie-9999::toaster'`,
 * the complete build log and the output of `emerge -pqv '=media-video/bcwc_pcie-9999::toaster'`.
 * Working directory: '/usr/lib64/python2.7/site-packages'
 * S: '/var/tmp/portage/media-video/bcwc_pcie-9999/work/bcwc_pcie-9999'
... done!

!!! All ebuilds that could satisfy "bcwc_pcie" have been masked.
!!! One of the following masked packages is required to complete your request:
- media-video/bcwc_pcie-9999::toaster (masked by: corruption)

For more information, see the MASKED PACKAGES section in the emerge
man page or refer to the Gentoo Handbook.
```

**Comment by toaster** [comment]
Argh … sorry … in the meantime the master branch got patched for 4.8.
I reverted the broken changes and switched back to master for now.
